hikari/impl/rest.py

```diff
@@ -791,7 +796,13 @@ async def fetch_channel(
         route = routes.GET_CHANNEL.compile(channel=channel)
         response = await self._request(route)
         assert isinstance(response, dict)
-        return self._entity_factory.deserialize_channel(response)
+        result = self._entity_factory.deserialize_channel(response)
+
+        if result.type is channels_.ChannelType.DM and self._cache:
+            assert isinstance(result, channels_.DMChannel)
+            self._cache.set_dm_channel_id(result.recipient.id, result.id)
```

Comment:
```suggestion
        if self._cache and isinstance(result, channels_.DMChannel):
            self._cache.set_dm_channel_id(result.recipient.id, result.id)
```

Reply:
Wouldn't that be slower for optimised runs for what is the exact same result?